src/pytest_split/plugin.py

```diff
@@ -134,4 +134,23 @@ def _calculate_suite_start_and_end_idx(splits, group, items, stored_durations):
     if not end_idx:
         end_idx = len(items)
 
+    # Modify start_idx and end_idx so they match start/end of an ipynb file
+    if _is_ipy_notebook(item_node_ids[start_idx]):
+        start_idx = _get_boundary_idx(item_node_ids, start_idx, "start")
+    if _is_ipy_notebook(item_node_ids[end_idx - 1]):  # since Python indexing is [a, b)
+        end_idx = _get_boundary_idx(item_node_ids, end_idx - 1, "end")
+
     return start_idx, end_idx
+
+
+def _is_ipy_notebook(node_id):
+    fpath = node_id.split("::")[0]
+    return True if fpath.endswith(".ipynb") else False
+
+
+def _get_boundary_idx(item_node_ids, idx, mode):
```

Comment:
As the mode is basically a flag (two possible values), I think boolean would be more typo-safe. E.g. `def _get_boundary_idx(item_node_ids, idx, *, start_mode)`